anomalib/utils/metrics/pro.py

```diff
@@ -0,0 +1,111 @@
+"""Implementation of PRO metric based on TorchMetrics."""
+from typing import List
+
+import cv2
+import numpy as np
+import torch
+from kornia.contrib import connected_components
+from torch import Tensor
+from torchmetrics import Metric
+from torchmetrics.functional import recall
+from torchmetrics.utilities.data import dim_zero_cat
+
+
+class PRO(Metric):
+    """Per-Region Overlap (PRO) Score."""
+
+    target: List[Tensor]
+    preds: List[Tensor]
+
+    def __init__(self, threshold: float = 0.5, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.threshold = threshold
+
+        self.add_state("preds", default=[], dist_reduce_fx="cat")
+        self.add_state("target", default=[], dist_reduce_fx="cat")
+
+    def update(self, predictions: Tensor, targets: Tensor) -> None:
+        """Compute the PRO score for the current batch."""
+
+        self.target.append(targets)
+        self.preds.append(predictions)
+
+    def compute(self) -> Tensor:
+        """Compute the macro average of the PRO score across all regions in all batches."""
+        target = dim_zero_cat(self.target)
+        preds = dim_zero_cat(self.preds)
+
+        if target.is_cuda:
+            comps = connected_components_gpu(target.unsqueeze(1))
+        else:
+            comps = connected_components_cpu(target.unsqueeze(1))
+        if not len(comps.unique()) > comps.max():
+            print("here")
+            connected_components_cpu(target.unsqueeze(1))
```

Comment:
Why `connected_components_cpu(target.unsqueeze(1))` doesn't return a value. I thought this would be something like;

```
comps = connected_components_cpu(target.unsqueeze(1))
```

Reply:
This was a leftover from debugging. Removed it now